### PR TITLE
 Allow the JWKClient to use a custom extractor

### PR DESCRIPTION
**Pull request by maxekman**
Also fixes shadowing of errors in `FromMultiple`.

### CODE REVIEW

**Comment by maxekman** [comment]
I found out the hard way that the JWKClient was hard coded to extract the secret from the HTTP header, which did not work when I used a custom extractor in the Validator.

File: jwk_client.go [hunk 1]
```diff
@@ -92,18 +99,18 @@ func (j *JWKClient) downloadKeys() ([]jose.JSONWebKey, error) {
 	return jwks.Keys, nil
 }
 
-func (j *JWKClient) GetSecret(req *http.Request) (interface{}, error) {
-	t, err := FromHeader(req)
-
+// GetSecret implements the GetSecret method of the SecretProvider interface.
+func (j *JWKClient) GetSecret(r *http.Request) (interface{}, error) {
```
Comment (maxekman) [inline jwk_client.go]: I did not want to break the interface here, but another option would be to let this get the token directly. That would avoid calling the extractor twice, once here and once in `Validate()`.

**Comment by yageek** [comment]
Could you just fix the `example/main.go` file also?

**Comment by maxekman** [comment]
Sure, missed that one. Maybe for the future, the example could me made into a test case? I could even give it a go now if you want?

**Comment by yageek** [comment]
Sure :) It would be great !

**Comment by maxekman** [comment]
I fixed the bug in the example. But looking at it I realize it's too complex to easily port to a example test, so I skipped that part.

**Comment by yageek** [comment]
Not an issue. Thanks for the work :)

**Pull request merged**